sklearn/neighbors/kd_tree.pyx

```diff
@@ -36,21 +36,53 @@ cdef class KDTree(BinaryTree):
 cdef int allocate_data(BinaryTree tree, ITYPE_t n_nodes,
                        ITYPE_t n_features) except -1:
     """Allocate arrays needed for the KD Tree"""
-    tree.node_bounds_arr = np.zeros((2, n_nodes, n_features), dtype=DTYPE)
+    cdef double d = 0
+    cdef float f = 0
+
+    if tree.is_float:
+        return _allocate_data(tree, n_nodes, n_features, f)
```

Comment:
Hmm... Okay. You should be able to cast a constant rather than have two local variables.